core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -926,7 +926,9 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     // The call to new NewHadoopJob automatically adds security credentials to conf,
     // so we don't need to explicitly add them ourselves
     val job = new NewHadoopJob(conf)
-    NewFileInputFormat.addInputPath(job, new Path(path))
+    // Use addInputPaths so that newAPIHadoopFile aligns with hadoopFile in taking
+    // comma separated files as input. (see SPARK-7155)
+    NewFileInputFormat.addInputPaths(job, path)
```

Comment:
Yes but originally to support multiple paths one would achieve it by adding them directly to the config. Code that was relying on it breaks (my case :() though it's easy to fix. In practice I am not sure there is any real advantage in preventing people to add paths from some code outside spark. WDYT?